common/djangoapps/course_modes/views.py

```diff
@@ -253,17 +253,28 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
         if deadline:
             formatted_audit_access_date = strftime_localized_html(deadline, 'SHORT_DATE')
             context['audit_access_deadline'] = formatted_audit_access_date
-        fbe_is_on = deadline and gated_content
+        full_fbe_is_on = deadline and gated_content
+        partial_fbe_is_on = deadline or gated_content
 
+        # Route to correct Track Selection page.
         # REV-2133 TODO Value Prop: remove waffle flag after testing is completed
         # and happy path version is ready to be rolled out to all users.
         if VALUE_PROP_TRACK_SELECTION_FLAG.is_enabled():
-            # First iteration of happy path does not handle errors. If there are enrollment errors for a learner that is
-            # technically considered happy path, old Track Selection page will be displayed.
-            if not error:
-                # Happy path conditions.
-                if verified_mode and fbe_is_on and not enterprise_customer:
-                    return render_to_response("course_modes/track_selection.html", context)
+            if not error:  # TODO: Remove by executing REV-2355
+                if not enterprise_customer:  # TODO: Remove by executing REV-2342
+                    if full_fbe_is_on:
+                        return render_to_response("course_modes/track_selection_types/full_fbe.html", context)
+                    elif partial_fbe_is_on:
+                        return render_to_response("course_modes/track_selection_types/partial_fbe.html", context)
```

Comment:
My understanding in conversation is that this 'non happy path' page should show as long as FBE is not fully on, so this could mean FBE partially enabled as you're checking here, or all-the-way off. Looking back at the ticket REV-2325 I see the confusion: It mentions FBE partially-on, but not FBE off. Let's parking lot this to confirm, in which case you won't need the partial_fbe_is_on variable: you can just go to the happy path page if full_fbe_is_on, otherwise go to this unhappy path page.

Reply:
Gotcha, thank you! I had assumed `partial_fbe_is_on` was required because of the second requirement bullet at the top of [REV-2325](https://openedx.atlassian.net/browse/REV-2325) ("In-scope Unhappy paths: [...] No verified upgrade (remove track selection)"). Looking forward to the parking lot.

Reply:
Result of parking lot: ticket has been amended so that the new screen in this ticket (currently named `partial_fbe.html`) will show when the enrollment is in one of the following cases:
1. course duration limits but not gated content
2. gated content but not course duration limits
3. neither course duration limit nor gated content

The page `full_fbe.html` will show instead of `partial_fbe.html` in the case that [4.] both course duration limit and gated content are enabled for the enrollment.

If the course does not have a `verified` mode, do not show `partial_fbe.html` or `full_fbe.html`. Instead, redirect to course page if the user has access to the course, or the dashboard if they do not yet have access to the course.

Rename HTML files for clarity.

To do:
- [ ] Rename `partial_fbe.html` to `nonfbe.html`
- [ ] Rename `full_fbe.html` to `fbe.html`
- [ ] Make case 3 above redirect to `nonfbe.html`
- [ ] Make case 4 above redirect to course/dashboard page.

Reply:
cool! And you've probably seen it as you make your changes locally, but the '3 cases' above can just be one case in the logic: 
if no course duration limits OR not gated content